File: python_examples/api_call.py
"""
Simple API Call using LangChain
Author: Optimum AI Lab
Description: This example demonstrates the simplest way to call an LLM
using LangChain to generate a response to a prompt.
"""

from langchain_openai import ChatOpenAI

# 1. Initialize the LLM model
llm = ChatOpenAI(model="gpt-4-turbo-preview")

# 2. Invoke the model with a prompt
if __name__ == "__main__":
    prompt = "Explain why the sky is blue in one sentence."
    print(f"Prompt: {prompt}")
    print("-" * 50)
    
    result = llm.invoke(prompt)
    print(f"Response: {result.content}")

File: java_examples/src/main/java/com/example/ApiCall.java
package com.example;

import dev.langchain4j.model.chat.ChatLanguageModel;
import dev.langchain4j.model.openai.OpenAiChatModel;

/**
 * Simple API Call using LangChain4j
 * Author: Optimum AI Lab
 * Description: This example demonstrates the simplest way to call an LLM
 * using LangChain4j to generate a response to a prompt.
 */
public class ApiCall {

    public static void main(String[] args) {
        System.out.println("Initializing LLM...");

        // 1. Build the chat model with your API key
        ChatLanguageModel model = OpenAiChatModel.builder()
                .apiKey(System.getenv("OPENAI_API_KEY"))
                .modelName("gpt-4-turbo-preview")
                .build();

        // 2. Generate a response to a prompt
        String prompt = "Explain why the sky is blue in one sentence.";
        System.out.println("Prompt: " + prompt);
        System.out.println("-".repeat(50));

        String response = model.generate(prompt);
        System.out.println("Response: " + response);
    }
}
